net/peer.go

```diff
@@ -28,3 +36,55 @@ func CreatePeer(addr string, tls bool) Peer {
 		tls:  tls,
 	}
 }
+
+// RemoteAddress returns the address of the peer by first taking the address
+// that gRPC returns. If that address is a reserved address, then it tries to
+// read the "X-REAL-IP" header content.
+// For example, a valid nging config could include
+//
+// ```
+// location / {
+//       grpc_pass grpc://127.0.0.1:9091;
+//       grpc_set_header X-Real-IP $remote_addr;
+// }
+// ```
+//
+func RemoteAddress(c context.Context) string {
+	peer, ok := peer.FromContext(c)
+	var str string = ""
+	if ok {
+		str = peer.Addr.String()
+	}
+	// https://en.wikipedia.org/wiki/Reserved_IP_addresses
+	reserved := []string{
```

Comment:
you're missing 172.16.
more generally, can crib off of https://github.com/multiformats/go-multiaddr-net/blob/master/private.go#L11